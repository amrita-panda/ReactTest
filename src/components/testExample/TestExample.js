import React, { Component } from 'react';
import {convert} from '../utils/Currency';
import {testExampleServices} from '../services/testExampleServices'

class TestExample extends Component {
    constructor() {
        super();
        this.state = {
          title: "",
          description: "",
          submitActive: false,
          data:{},
          data2:{}
        };
      }
       componentDidMount(){
            convert('USD','CAD').then(response=>{
              console.log("currency api",response);
            }).catch(error=>{
              console.log(error);
            })

           testExampleServices.getUsers().then(response2=>{
             console.log("user api",response2)
           }).catch(error=>{
             console.log(error);
           })
            
       }
      handleChange = event => {
        this.setState({
          [event.target.name]: event.target.value
        },this.checkFields);
      };
      handleSubmit = event => {
        event.preventDefault();
      
        if (!this.state.submitActive) {
          return
        }
        const submission = {
          title: this.state.title,
          description: this.state.description
        };
        console.log(this.props.submit)
        this.props.submit(submission);
        this.setState({
          title: '',
          description: '',
          submitActive: 'false'
        })
      };
      checkFields = () => {
        if (this.state.title && this.state.description) {
          this.setState({submitActive: true});
        } else {
          this.setState({submitActive: false});
        }
      };
      fetchData=() =>{
       convert('USD','CAD').then(response=>{
         console.log(response);
         this.setState({data:response})
       })
    }
      render() {
       // console.log(this.state.data)
        return (
          <form 
            className="controlled-form"
            onSubmit={event => this.handleSubmit(event)}
          >
            <input 
              className="text-input title-input"
              type="text"
              name="title"
              value={this.state.title}
              onChange={event => {
                this.handleChange(event);  
              }}
            />
            <input 
              className="text-input description-input"
              type="text"
              name="description"
              value={this.state.description}
              onChange={event => {
                this.handleChange(event);  
              }}
            />
            <button 
              className={`submit-btn ${this.state.submitActive}`}
            >submit</button>
            <button type='button'id='apifetchData-id' onClick={()=>this.fetchData()}>Fetch Data</button>
          </form>
        );
      }
    }
export default TestExample;