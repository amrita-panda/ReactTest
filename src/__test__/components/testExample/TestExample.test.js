import { mount, shallow } from 'enzyme';
import moxios from 'moxios';
import React from 'react';
import TestExample from '../../../components/testExample/TestExample';

describe('controlled form testExample',()=>{
    let wrapper;
    let mockSubmit;

    beforeEach(()=>{
        mockSubmit=jest.fn();
        wrapper=shallow(<TestExample submit={mockSubmit}/>)
    })
    it('should match the snapshot',()=>{
        expect(wrapper).toMatchSnapshot();
    })
})

describe('handelChange',()=>{
    let wrapper;
    let mockSubmit;

    beforeEach(()=>{
        mockSubmit=jest.fn();
        wrapper=shallow(<TestExample submit={mockSubmit}/>)
    })
    it('should call setState on title',()=>{
        const mockEvent={
            target:{
                name:"title",
                value:'testing Title'
            }
        };
        const expected={
            title: 'testing Title',
            description: '',
            submitActive: false,
            data:[]
        }
        wrapper.instance().handleChange(mockEvent);
        console.log(wrapper.state())
        expect(wrapper.state()).toEqual(expected)
    })

    it('should call checkFields',()=>{
        const spy=jest.spyOn(wrapper.instance(),'checkFields')
        wrapper.instance().forceUpdate();
        const mockEvent={
            target:{
                name:"description",
                value:"testing"
            }
        };
        const expected=true;
        wrapper.instance().handleChange(mockEvent);
        expect(spy).toHaveBeenCalled();
    })

    it('should call handelChange on description change with the correct params',()=>{
        const spy=jest.spyOn(wrapper.instance(),"handleChange");
        wrapper.instance().forceUpdate();
        const mockEvent={
            target:{
                name:"description",
                value:"testing",
            }
        };
        wrapper.find(".description-input").simulate("change",mockEvent)
        expect(spy).toHaveBeenCalledWith(mockEvent);

    })

    it('it should call preventDefault',()=>{
        const mockPreventDefault=jest.fn();
        const mockEvent={
            preventDefault:mockPreventDefault
        };
        console.log("hi",mockPreventDefault);
        console.log("mockevent",mockEvent)
        wrapper.instance().handleSubmit(mockEvent);
        expect(mockPreventDefault).toHaveBeenCalled()
    })
})

describe('fetchapi',()=>{
    let wrapped;
    beforeEach(()=>{
        wrapped=mount(<TestExample></TestExample>)
        // moxios.install();
    })
    afterEach(()=>{
        // moxios.uninstall();
    })
    it('should call a api to get all the comments',()=>{
        let mockResponse=[
            {postId: 1, id: 1, name: "id labore ex et quam laborum", email: "Eliseo@gardner.biz", body: "laudantium enim quasi est quidem magnam voluptate …utem quasi↵reiciendis et nam sapiente accusantium"}
        ]
         const spy=jest.spyOn(wrapped.instance(),"fetchData");
       const fetchButton= wrapped.find('#apifetchData-id').simulate('click');
       expect(fetchButton.length).toEqual(1);
       expect(spy).toBeCalled()
       wrapped.update();
       
       
       
  

        
    })

      
       
        
    })