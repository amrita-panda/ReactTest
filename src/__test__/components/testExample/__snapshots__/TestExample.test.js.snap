// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`controlled form testExample should match the snapshot 1`] = `ShallowWrapper {}`;
